taichi/codegen/spirv/spirv_codegen.cpp

```diff
@@ -1830,10 +1834,24 @@ class TaskCodegen : public IRVisitor {
 
     std::vector<std::tuple<spirv::SType, std::string, size_t>>
         struct_components_;
+    // Now we only have one ret
     for (auto &ret : ctx_attribs_->rets()) {
-      struct_components_.emplace_back(ir_->get_primitive_type(ret.dt),
-                                      "ret" + std::to_string(ret.index),
-                                      ret.offset_in_mem);
+      if (auto tensor_type = ret.dt->cast<TensorType>()) {
+        for (int i = 0; i < tensor_type->get_num_elements(); i++) {
+          struct_components_.emplace_back(
```

Comment:
I don't quite understand this bit, wonder why there are two members for each element of the tensor?

Reply:
Just for unifying the memory form with other backends. LLVM backend uses i64 data type to store return values, but Spir-v only support 32bit data type, so I use two i32 slots to represent 64bit.

Reply:
Hello, please generate the proper types. 64 bit data types are supported in SPIR-V.

Reply:
In tensor types, you should get the primitive type of the tensor, and generate a proper array type, then put the array type under the ret struct

Reply:
Fix